### DISCUSSION TITLE
Wrapping text at 80 characters

### DISCUSSION THREAD

**Discussion by mkemorgn** [LazyVim/LazyVim#2324] [Q&A]
I've seen a few of these kind of discussions but haven't found a solution that works.  I tried setting `opt.textwidth=80` and that didn't change anything no matter what kind of file it is.  

In the editor when I type `:set textwidth?` the editor returns with '80' but lines still aren't being wrapped at 80 characters.  

UPDATE:
Even when I set `opt.wrap = true` the lines don't wrap at 80 characters.  They wrap at the edge of the window.

**Comment by dpetka2001** [comment]
That's because `vim.opts.wrap = false` [here](https://github.com/LazyVim/LazyVim/blob/879e29504d43e9f178d967ecc34d482f902e5a91/lua/lazyvim/config/options.lua#L59). In your `options.lua` you would have to set `vim.opts.wrap = true` or you can also toggle wrapping with the keymap `<leader>uw`.

**Reply by mkemorgn** [reply]
I should've added that I set `opt.wrap = true`.  Lines wrap at the end of the window but not at 80 characters.

**Reply by dpetka2001** [reply]
It must be because of the formatter. Look to see in the formatter options that you use, the option for maximum length line. 
For example for lua files, `stylua` the formatter is responsible for this, so you have to make the change in the `stylua.toml` file and the name of the option is `column_width`.

**Comment by jackplus-xyz** [comment]
Hi @mkemorgn , were you able to  fix this problem? It seems like word warp weren't working correctly, it truncates the line at the character in the end of the window instead of the word.

I've tried these but none of it worked:
- Turn on word wrapping by `<leader>uw`
- `:set textwidth=80`
- change the `column_width` in `stylua.toml` to 80

<img width="1392" alt="Screenshot 2024-01-30 at 20 59 13" src="https://github.com/LazyVim/LazyVim/assets/103010556/db594409-5a52-459c-8a1d-7f7deaad206b">

**Reply by dpetka2001** [reply]
You should look into `:h linebreak` if you want the wrapping to occur at a word boundary.

**Reply by jackplus-xyz** [reply]
Thanks. That works!

**Reply by hinricht** [reply]
@jackplus-xyz Could you please post your solution ?
